pkg/os/os.go

```diff
@@ -91,3 +93,52 @@ func (RealOS) WriteFile(filename string, data []byte, perm os.FileMode) error {
 func (RealOS) Hostname() (string, error) {
 	return os.Hostname()
 }
+
+// InflightStat represents a Stat call that is in progress for a given path.
+// It is exported so that consumers of the OS interface can use it to coordinate.
+type InflightStat struct {
+	Cond *sync.Cond // Used to signal completion to waiting goroutines.
+	Done bool       // Flag indicating if the Stat call is complete.
+	Err  error      // Resulting error from Stat.
+}
+
+type StatResult struct {
+	Err error
+}
+
+type Manager interface {
```

Comment:
`Manager` seems a bit general, are there other use cases besides `stat` for this type? If not, I'd prefer to keep this type more focused and we can refactor later if that changes (e.g. Change this to `StatManager` or something of that sort).